<a href="https://colab.research.google.com/github/Sandeep1098-pil/CaptureMoment/blob/main/screen_time.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Importing Libraries**

In [28]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

**Loading Dataset**

In [29]:
from google.colab import files
uploaded = files.upload()


Saving Indian_Kids_Screen_Time.csv to Indian_Kids_Screen_Time (1).csv


In [30]:
df=pd.read_csv("Indian_Kids_Screen_Time.csv")
df

Age  Gender  Avg_Daily_Screen_Time_hr Primary_Device  \
0      14    Male                      3.99     Smartphone   
1      11  Female                      4.61         Laptop   
2      18  Female                      3.73             TV   
3      15  Female                      1.21         Laptop   
4      12  Female                      5.89     Smartphone   
...   ...     ...                       ...            ...   
9707   17    Male                      3.26     Smartphone   
9708   17  Female                      4.43     Smartphone   
9709   16    Male                      5.62     Smartphone   
9710   17    Male                      5.60             TV   
9711   15  Female                      6.12             TV   

      Exceeded_Recommended_Limit  Educational_to_Recreational_Ratio  \
0                           True                               0.42   
1                           True                               0.30   
2                           True                               0.32   
3                          False                               0.39   
4                           True                               0.49   
...                          ...                                ...   
9707                        True                               0.44   
9708                        True                               0.40   
9709                        True                               0.39   
9710                        True                               0.43   
9711                        True                               0.33   

                       Health_Impacts Urban_or_Rural  
0              Poor Sleep, Eye Strain          Urban  
1                          Poor Sleep          Urban  
2                          Poor Sleep          Urban  
3                                 NaN          Urban  
4                 Poor Sleep, Anxiety          Urban  
...                               ...            ...  
9707                       Poor Sleep          Urban  
9708                       Poor Sleep          Rural  
9709  Poor Sleep, Eye Strain, Anxiety          Rural  
9710                       Poor Sleep          Urban  
9711                          Anxiety          Urban  

[9712 rows x 8 columns]

Understanding data

In [31]:
df.head()

Age  Gender  Avg_Daily_Screen_Time_hr Primary_Device  \
0   14    Male                      3.99     Smartphone   
1   11  Female                      4.61         Laptop   
2   18  Female                      3.73             TV   
3   15  Female                      1.21         Laptop   
4   12  Female                      5.89     Smartphone   

   Exceeded_Recommended_Limit  Educational_to_Recreational_Ratio  \
0                        True                               0.42   
1                        True                               0.30   
2                        True                               0.32   
3                       False                               0.39   
4                        True                               0.49   

           Health_Impacts Urban_or_Rural  
0  Poor Sleep, Eye Strain          Urban  
1              Poor Sleep          Urban  
2              Poor Sleep          Urban  
3                     NaN          Urban  
4     Poor Sleep, Anxiety          Urban

In [32]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9712 entries, 0 to 9711
Data columns (total 8 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   Age                                9712 non-null   int64  
 1   Gender                             9712 non-null   object 
 2   Avg_Daily_Screen_Time_hr           9712 non-null   float64
 3   Primary_Device                     9712 non-null   object 
 4   Exceeded_Recommended_Limit         9712 non-null   bool   
 5   Educational_to_Recreational_Ratio  9712 non-null   float64
 6   Health_Impacts                     6494 non-null   object 
 7   Urban_or_Rural                     9712 non-null   object 
dtypes: bool(1), float64(2), int64(1), object(4)
memory usage: 540.7+ KB


In [33]:
df.columns

Index(['Age', 'Gender', 'Avg_Daily_Screen_Time_hr', 'Primary_Device',
       'Exceeded_Recommended_Limit', 'Educational_to_Recreational_Ratio',
       'Health_Impacts', 'Urban_or_Rural'],
      dtype='object')

In [34]:
df.describe()

Age  Avg_Daily_Screen_Time_hr  \
count  9712.000000               9712.000000   
mean     12.979201                  4.352837   
std       3.162437                  1.718232   
min       8.000000                  0.000000   
25%      10.000000                  3.410000   
50%      13.000000                  4.440000   
75%      16.000000                  5.380000   
max      18.000000                 13.890000   

       Educational_to_Recreational_Ratio  
count                        9712.000000  
mean                            0.427226  
std                             0.073221  
min                             0.300000  
25%                             0.370000  
50%                             0.430000  
75%                             0.480000  
max                             0.600000

**Data Cleaning**

In [35]:
#checking null values
df.isnull().sum()

Age                                     0
Gender                                  0
Avg_Daily_Screen_Time_hr                0
Primary_Device                          0
Exceeded_Recommended_Limit              0
Educational_to_Recreational_Ratio       0
Health_Impacts                       3218
Urban_or_Rural                          0
dtype: int64

In [36]:
#checking duplicate values
df[df.duplicated()]

Age  Gender  Avg_Daily_Screen_Time_hr Primary_Device  \
996    16  Female                      4.42         Laptop   
1700    9  Female                      0.00     Smartphone   
2090   10  Female                      0.00         Tablet   
3089   16    Male                      4.81     Smartphone   
3213    8  Female                      4.84     Smartphone   
3308    8  Female                      0.00     Smartphone   
3572    8    Male                      0.00     Smartphone   
4166    8  Female                      0.00     Smartphone   
4460    8  Female                      0.00     Smartphone   
4960   10  Female                      0.00     Smartphone   
5070    8  Female                      0.00     Smartphone   
5073    9  Female                      0.00             TV   
5291    8  Female                      0.00     Smartphone   
5892   11    Male                      3.86             TV   
6041   11  Female                      3.20     Smartphone   
6046   10    Male                      0.00     Smartphone   
6175    9  Female                      0.00     Smartphone   
6422    9    Male                      0.00     Smartphone   
6880   17  Female                      4.93         Laptop   
6927    9  Female                      0.00     Smartphone   
7018   10  Female                      0.00     Smartphone   
7103    9    Male                      0.00         Tablet   
7219    9  Female                      0.00     Smartphone   
7265   10  Female                      0.00             TV   
7633   10  Female                      0.00         Tablet   
7720   11  Female                      2.74     Smartphone   
7839    8  Female                      0.00     Smartphone   
7945   10    Male                      0.00     Smartphone   
8045   10  Female                      0.00     Smartphone   
8075   14    Male                      4.94     Smartphone   
8127    9    Male                      0.00     Smartphone   
8264   10    Male                      0.00         Tablet   
8297    8  Female                      0.00     Smartphone   
8829    9    Male                      0.00             TV   
8896    8  Female                      0.00             TV   
8932    8  Female                      1.82         Tablet   
8991   10  Female                      1.07             TV   
9018    8  Female                      0.00     Smartphone   
9457   10  Female                      0.00     Smartphone   
9471    9  Female                      4.44             TV   
9517   10  Female                      0.00     Smartphone   
9532   18    Male                      5.98     Smartphone   
9633   10    Male                      0.00             TV   
9641   10    Male                      0.00             TV   

      Exceeded_Recommended_Limit  Educational_to_Recreational_Ratio  \
996                         True                               0.35   
1700                       False                               0.60   
2090                       False                               0.49   
3089                        True                               0.31   
3213                        True                               0.46   
3308                       False                               0.53   
3572                       False                               0.41   
4166                       False                               0.47   
4460                       False                               0.47   
4960                       False                               0.55   
5070                       False                               0.59   
5073                       False                               0.58   
5291                       False                               0.49   
5892                        True                               0.30   
6041                        True                               0.35   
6046                       False                               0.45   
6175    

In [37]:
#replacing null values with most repeating value
df['Health_Impacts']=df['Health_Impacts'].fillna(df['Health_Impacts'].mode()[0])
df.isnull().sum()

Age                                  0
Gender                               0
Avg_Daily_Screen_Time_hr             0
Primary_Device                       0
Exceeded_Recommended_Limit           0
Educational_to_Recreational_Ratio    0
Health_Impacts                       0
Urban_or_Rural                       0
dtype: int64

**Checking outliers using IQR method**

In [38]:
def count_outliers_iqr(df, column):
    Q1 = df[column].quantile(0.25)
    Q3 = df[column].quantile(0.75)
    IQR = Q3 - Q1
    lower = Q1 - 1.5 * IQR
    upper = Q3 + 1.5 * IQR
    outliers = df[(df[column] < lower) | (df[column] > upper)]
    print(f"{column} → {len(outliers)} outliers")
    return outliers


count_outliers_iqr(df, 'Avg_Daily_Screen_Time_hr')
count_outliers_iqr(df, 'Age')
count_outliers_iqr(df, 'Educational_to_Recreational_Ratio')

Avg_Daily_Screen_Time_hr → 448 outliers
Age → 0 outliers
Educational_to_Recreational_Ratio → 0 outliers


Empty DataFrame
Columns: [Age, Gender, Avg_Daily_Screen_Time_hr, Primary_Device, Exceeded_Recommended_Limit, Educational_to_Recreational_Ratio, Health_Impacts, Urban_or_Rural]
Index: []

we can see there are 448 outliers in Avg_Daily_Screen_Time_hr attribute

In [40]:
#drop outliers

def drop_outliers_iqr(df, column):
    Q1 = df[column].quantile(0.25)
    Q3 = df[column].quantile(0.75)
    IQR = Q3 - Q1
    lower = Q1 - 1.5 * IQR
    upper = Q3 + 1.5 * IQR
    return df[(df[column] >= lower) & (df[column] <= upper)]
df = drop_outliers_iqr(df, 'Avg_Daily_Screen_Time_hr')
df.shape

(9174, 8)

Feature Engineering

In [41]:
#converting male and female into 0 and 1
df['Gender_binary'] = df['Gender'].map({'Male': 1, 'Female': 0})

In [42]:
df

Age  Gender  Avg_Daily_Screen_Time_hr Primary_Device  \
0      14    Male                      3.99     Smartphone   
1      11  Female                      4.61         Laptop   
2      18  Female                      3.73             TV   
3      15  Female                      1.21         Laptop   
4      12  Female                      5.89     Smartphone   
...   ...     ...                       ...            ...   
9707   17    Male                      3.26     Smartphone   
9708   17  Female                      4.43     Smartphone   
9709   16    Male                      5.62     Smartphone   
9710   17    Male                      5.60             TV   
9711   15  Female                      6.12             TV   

      Exceeded_Recommended_Limit  Educational_to_Recreational_Ratio  \
0                           True                               0.42   
1                           True                               0.30   
2                           True                               0.32   
3                          False                               0.39   
4                           True                               0.49   
...                          ...                                ...   
9707                        True                               0.44   
9708                        True                               0.40   
9709                        True                               0.39   
9710                        True                               0.43   
9711                        True                               0.33   

                       Health_Impacts Urban_or_Rural  Gender_binary  
0              Poor Sleep, Eye Strain          Urban              1  
1                          Poor Sleep          Urban              0  
2                          Poor Sleep          Urban              0  
3                          Poor Sleep          Urban              0  
4                 Poor Sleep, Anxiety          Urban              0  
...                               ...            ...            ...  
9707                       Poor Sleep          Urban              1  
9708                       Poor Sleep          Rural              0  
9709  Poor Sleep, Eye Strain, Anxiety          Rural              1  
9710                       Poor Sleep          Urban              1  
9711                          Anxiety          Urban              0  

[9174 rows x 9 columns]

In [43]:
#creating attribute urban and rural into binary
df['Urban_binary'] = df['Urban_or_Rural'].map({'Urban': 1, 'Rural': 0})
df

Age  Gender  Avg_Daily_Screen_Time_hr Primary_Device  \
0      14    Male                      3.99     Smartphone   
1      11  Female                      4.61         Laptop   
2      18  Female                      3.73             TV   
3      15  Female                      1.21         Laptop   
4      12  Female                      5.89     Smartphone   
...   ...     ...                       ...            ...   
9707   17    Male                      3.26     Smartphone   
9708   17  Female                      4.43     Smartphone   
9709   16    Male                      5.62     Smartphone   
9710   17    Male                      5.60             TV   
9711   15  Female                      6.12             TV   

      Exceeded_Recommended_Limit  Educational_to_Recreational_Ratio  \
0                           True                               0.42   
1                           True                               0.30   
2                           True                               0.32   
3                          False                               0.39   
4                           True                               0.49   
...                          ...                                ...   
9707                        True                               0.44   
9708                        True                               0.40   
9709                        True                               0.39   
9710                        True                               0.43   
9711                        True                               0.33   

                       Health_Impacts Urban_or_Rural  Gender_binary  \
0              Poor Sleep, Eye Strain          Urban              1   
1                          Poor Sleep          Urban              0   
2                          Poor Sleep          Urban              0   
3                          Poor Sleep          Urban              0   
4                 Poor Sleep, Anxiety          Urban              0   
...                               ...            ...            ...   
9707                       Poor Sleep          Urban              1   
9708                       Poor Sleep          Rural              0   
9709  Poor Sleep, Eye Strain, Anxiety          Rural              1   
9710                       Poor Sleep          Urban              1   
9711                          Anxiety          Urban              0   

      Urban_binary  
0                1  
1                1  
2                1  
3                1  
4                1  
...            ...  
9707             1  
9708             0  
9709             0  
9710             1  
9711             1  

[9174 rows x 10 columns]

In [44]:
#creating attribute with 0 and 1 values by Exceeded_Recommended_Limit
df['Exceeded_Recommended_Limit_binary'] = df['Exceeded_Recommended_Limit'].map({True: 1, False: 0})
df

Age  Gender  Avg_Daily_Screen_Time_hr Primary_Device  \
0      14    Male                      3.99     Smartphone   
1      11  Female                      4.61         Laptop   
2      18  Female                      3.73             TV   
3      15  Female                      1.21         Laptop   
4      12  Female                      5.89     Smartphone   
...   ...     ...                       ...            ...   
9707   17    Male                      3.26     Smartphone   
9708   17  Female                      4.43     Smartphone   
9709   16    Male                      5.62     Smartphone   
9710   17    Male                      5.60             TV   
9711   15  Female                      6.12             TV   

      Exceeded_Recommended_Limit  Educational_to_Recreational_Ratio  \
0                           True                               0.42   
1                           True                               0.30   
2                           True                               0.32   
3                          False                               0.39   
4                           True                               0.49   
...                          ...                                ...   
9707                        True                               0.44   
9708                        True                               0.40   
9709                        True                               0.39   
9710                        True                               0.43   
9711                        True                               0.33   

                       Health_Impacts Urban_or_Rural  Gender_binary  \
0              Poor Sleep, Eye Strain          Urban              1   
1                          Poor Sleep          Urban              0   
2                          Poor Sleep          Urban              0   
3                          Poor Sleep          Urban              0   
4                 Poor Sleep, Anxiety          Urban              0   
...                               ...            ...            ...   
9707                       Poor Sleep          Urban              1   
9708                       Poor Sleep          Rural              0   
9709  Poor Sleep, Eye Strain, Anxiety          Rural              1   
9710                       Poor Sleep          Urban              1   
9711                          Anxiety          Urban              0   

      Urban_binary  Exceeded_Recommended_Limit_binary  
0                1                                  1  
1                1                                  1  
2                1                                  1  
3                1                                  0  
4                1                                  1  
...            ...                                ...  
9707             1                                  1  
9708             0                                  1  
9709             0                                  1  
9710             1                                  1  
9711             1                                  1  

[9174 rows x 11 columns]

In [45]:
df.shape

(9174, 11)

In [49]:
for col in ['Gender', 'Primary_Device', 'Exceeded_Recommended_Limit', 'Urban_or_Rural', 'Health_Impacts','Urban_binary','Gender_binary']:
    print(f"\nUnique values in '{col}':")
    print(df[col].unique())



Unique values in 'Gender':
['Male' 'Female']

Unique values in 'Primary_Device':
['Smartphone' 'Laptop' 'TV' 'Tablet']

Unique values in 'Exceeded_Recommended_Limit':
[ True False]

Unique values in 'Urban_or_Rural':
['Urban' 'Rural']

Unique values in 'Health_Impacts':
['Poor Sleep, Eye Strain' 'Poor Sleep' 'Poor Sleep, Anxiety'
 'Poor Sleep, Obesity Risk' 'Eye Strain' 'Obesity Risk' 'Anxiety'
 'Poor Sleep, Anxiety, Obesity Risk' 'Eye Strain, Obesity Risk'
 'Eye Strain, Anxiety, Obesity Risk' 'Anxiety, Obesity Risk'
 'Poor Sleep, Eye Strain, Obesity Risk' 'Poor Sleep, Eye Strain, Anxiety'
 'Poor Sleep, Eye Strain, Anxiety, Obesity Risk' 'Eye Strain, Anxiety']

Unique values in 'Urban_binary':
[1 0]

Unique values in 'Gender_binary':
[1 0]


In [50]:
df.to_csv('cleaned_data.csv', index=False)
from google.colab import files
files.download("cleaned_data.csv")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

As the dashboard is created by using power bi with the cleaned dataset